### PR TITLE
CORE-16860 counterparty info refactor

### PR DESCRIPTION
**Pull request by LWogan**
To request counterparty info when it has not already been exchanged, send a SessionCounterpartyInfoRQ.
Include SessionInit payload to start the remote flow. 
If data has already been exchanged SessionCounterpartyInfoRQ wont be sent.
Respond with SessionCounterpartyInfoRS session event payload. Blank fields as contextSessionProperties are set on SessionEvent 
SequenceNumber will be set to null for both payload types. Only SessionData and SessionClose payloads will have sequence numbers set on SessionEvent

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/SessionEvent.avsc [hunk 1]
```diff
@@ -50,8 +50,8 @@
     {
       "name": "payload",
       "type": [
-        "net.corda.data.flow.event.session.SessionInit",
```
Comment (mbrkic-r3) [inline data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/SessionEvent.avsc:53]: Do we need to remove https://github.com/corda/corda-api/blob/CORE-16860/counterparty-info/data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/session/SessionInit.avsc as well?

File: data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/FlowEvent.avsc [hunk 2]
```diff
@@ -11,7 +11,6 @@
       "name": "payload",
       "type": [
         "net.corda.data.flow.event.StartFlow",
-        "net.corda.data.flow.event.Wakeup",
```
Comment (owenstanford) [inline data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/FlowEvent.avsc:14]: It's a remote edge case, but removing types like this could be a problem when upgrading from 5.0 to 5.1, need to check with @simon-johnson-r3, also worth adding an avro upgrade test to validate the change (see existing tests for an example)

File: data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/SessionEvent.avsc (hunk 1 above)
Comment (owenstanford) [inline data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/SessionEvent.avsc:53]: same as above

File: data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/FlowEvent.avsc (hunk 2 above)
Comment (JamesHR3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/FlowEvent.avsc:14]: I think upgrade guarantees are just that you won't need to recompile your application. In particular I don't think we're guaranteeing that in flight flows will work across an upgrade, so any events of these types present in the system will need to be cleared before the upgrade can go ahead.

**Review by conal-smith — APPROVED** [review]

**Review by JamesHR3 — APPROVED** [review]
